src/main/java/com/clouway/cuse/spi/SearchEngineImpl.java

```diff
@@ -26,16 +27,24 @@ public SearchEngineImpl(EntityLoader entityLoader,
   }
 
   public void register(Object instance) {
-
     Class instanceClass = instance.getClass();
     IndexingStrategy strategy = indexingStrategyCatalog.get(instanceClass);
-
     if (strategy == null) {
       throw new NotConfiguredIndexingStrategyException();
     }
-
     indexRegistry.register(instance, strategy);
+  }
 
+  public void registerAll(List<? extends Object> objects) {
+    if (objects.isEmpty()) {
+      throw new EmptyIndexListException();
```

Comment:
I think that it will be better behavior if do nothing in this case, actually the list is not null but the there are no indexes that have to be handled in it